odoo/addons/base/models/ir_qweb.py

```diff
@@ -148,31 +722,880 @@ def is_child_view(view_name):
             for node in view:
                 if node.get('t-name'):
                     node.set('t-name', str(name))
+                    node.attrib.pop('name', None)
+                    node.attrib.pop('id', None)
             return (view, view_id)
         else:
             return (template, view_id)
 
+    # values for running time
+
+    def _prepare_values(self, values, options):
+        """ Prepare the context that will sent to the compiled and evaluated
+        function.
+
+        :param values: template values to be used for rendering
+        :param options: frozen dict of compilation parameters.
+        """
+        check_values(values)
+        values['true'] = True
+        values['false'] = False
+        if 'request' not in values:
+            values['request'] = request
+        return values
+
+    def _prepare_globals(self, globals_dict, options):
+        """ Prepare the global context that will sent to eval the qweb generated
+        code.
+
+        :param globals_dict: template global values use in compiled code
+        :param options: frozen dict of compilation parameters.
+        """
+        globals_dict['Sized'] = Sized
+        globals_dict['Mapping'] = Mapping
+        globals_dict['Markup'] = Markup
+        globals_dict['escape'] = escape
+        globals_dict['VOID_ELEMENTS'] = VOID_ELEMENTS
+        globals_dict['compile_options'] = options
+        globals_dict.update(AVAILABLE_OBJECTS)
+        return globals_dict
+
+    # helpers for compilation
+
+    def _append_text(self, text, options):
+        """ Add an item (converts to a string) to the list.
+            This will be concatenated and added during a call to the
+            `_flush_text` method. This makes it possible to return only one
+            yield containing all the parts."""
+        options['_text_concat'].append(self._compile_to_str(text))
+
+    def _flush_text(self, options, indent):
+        """Concatenate all the textual chunks added by the `_append_text`
+            method into a single yield."""
+        text_concat = options['_text_concat']
+        if text_concat:
+            text = ''.join(text_concat)
+            text_concat.clear()
+            return [f"{'    ' * indent}yield {repr(text)}"]
+        else:
+            return []
+
+    def _indent(self, code, indent):
+        """Indent the code to respect the python syntax."""
+        return _indent(code, '    ' * indent)
```

Comment:
Is this expected to be overridden?
IMHO this one and many others should simply be helper functions instead of methods.

I also noticed that you often use the same plumbing code over and over again, maybe you could abstract it?
`self._indent(dedent(...).strip(), indent)` → `self._indent_literal(..., indent)`